Fix Amount Range issue with Minterm and Add check for total sum

maxxyh opened the pull request:
Fix check for Amount(BigDecimal) to use 
cents range instead of dollars range 

Add try-catch to check if new expense/edited 
expense/new budget value will exceed MAX_VALUE 
of Amount class.

Fixes #171

maxxyh committed:
Fix Amount Range issue with Minterm and Add check for total sum

Fix check for Amount(BigDecimal) to use 
cents range instead of dollars range 

Add try-catch to check if new expense/edited 
expense/new budget value will exceed MAX_VALUE 
of Amount class.

maxxyh committed:
Fix checkstyle

lebencwb reviewed (approved):
LGTM! Thanks for fixing this!

maxxyh committed:
Create tallyExpenses() in Model, add tests for sum and budget overflow

Add tests for sum overflow for Edit, Add and Topup methods 

Add tallyExpenses() in Model for Testing
